src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.Translate/Main.cs

```diff
@@ -0,0 +1,193 @@
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// sd
+        /// </summary>
+        /// <param name="langs">Array of languages</param>
+        /// <returns>A converted list of 'language' to 'result'</returns>
+        public List<Result> ConvertArrayOfLanguagesToResult(Language[] langs)
+        {
+            var results = new List<Result>();
+
+            foreach (var lang in langs)
+            {
+                results.Add(new Result
+                {
+                    Title = lang.Code,
+                    SubTitle = $"{lang.DisplayName} - {lang.Name}",
+                    IcoPath = _iconPath,
+                    Action = (_) =>
+                    {
+                        Console.WriteLine("Action");
+                        return true;
+                    },
+                });
+            }
+
+            return results;
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            var arguments = query?.Search.Split(' ', 2);
+
+            var languageSearch = arguments?[0] ?? string.Empty;
+
+            var langs = LanguageHelper.FindLanguages(languageSearch);
+            var suggestedLangs = ConvertArrayOfLanguagesToResult(langs);
+
+            var targetText = (arguments?.Length > 1 && !string.IsNullOrWhiteSpace(arguments[1])) ? arguments[1] : null;
+            if (langs.Length > 0 && targetText is not null)
+            {
+                var translation = _translator.TranslateAsync(targetText, langs.First().Code);
+                var completed = translation.Wait(5000);
+
+                ITranslationResult? translationResult = null;
+                if (completed)
+                {
+                    translationResult = translation.Result;
+                }
+
+                var translatedText = translationResult?.Translation ?? Resources.TranslateError;
+
+                return new List<Result>()
+                {
+                    new Result
+                    {
+                        IcoPath = _iconPath,
+                        Title = translatedText,
+                        QueryTextDisplay = translatedText,
+                        SubTitle = Resources.CopyTranslation,
```

Comment:
Let's show the translation service used. 

Btw, which one do we use here inthe plugin? Can this be an input parameter* or setting (later possible via drop-down setting)?

*) `google:de My text` and `en My text`

Reply:
Translate service is automatically selected by GTranslate

There is something like a list of supported languages for each service, so it switches itself

I would like to be able to choose a service, but I do not know how to do it. It seems like you can only do one **ActionKeyword** to trigger this plugin

Reply:
> Let's show the translation service used

It's in ToolTipData

Reply:
I missed the tool tip thibg. It's okay for me to have it in the tooltip.